libs/api/domains/auth/src/lib/resolvers/delegation.resolver.ts

```diff
@@ -49,8 +49,14 @@ export class DelegationResolver {
   }
 
   @Query(() => [Delegation], { name: 'authDelegations' })
-  getDelegations(@CurrentUser() user: User): Promise<DelegationDTO[]> {
-    return this.authService.getDelegations(user)
+  async getDelegations(@CurrentUser() user: User): Promise<DelegationDTO[]> {
+    const delegations = await this.authService.getDelegations(user)
+    return delegations.filter(
+      (delegation) =>
+        delegation.scopes &&
+        delegation.scopes.length > 0 &&
```

Comment:
We also need to somewhere filter out individual scopes that are not valid anymore. Do you plan to fix that in another PR?

Reply:
I really feel like this shouldn't be handled by the resolver. To me, the resolver is just a consumer of the business logic that is implemented in `auth-(public-?)api`.

Same applies to the filtering that I did.

So IMO, this should be handled in auth-api.

Reply:
You're right. Can we move this there? :)